tests/containers/skopeo.pm

```diff
@@ -0,0 +1,119 @@
+    # skopeo --version | cut -d" " -f3 --> separates "skopeo version 1.12.0" output into and array and takes third value "1.12.0"
+
+    # Create test directory
+    script_run("mkdir /tmp/test");
+
+    ######### Inspect a docker repository
+
+    # assert_script_run --> runs the command and die unless it returns zero, indicating successful completion of $cmd
+    assert_script_run("skopeo inspect $remote_image",
+        fail_message => 'Failed to inspect remote image.');
+
+
+    ######### Pull the image into a directory
+
+
+    validate_script_output("skopeo copy $remote_image dir:$workdir",
+        sub { m/Writing manifest to image destination/ },
+        fail_message => 'Failed to copy image.');
+
+    # Unpacked contents must include a manifest and version
+    validate_script_output("file $workdir/manifest.json",
+        sub { $_ !~ m/cannot/ },
+        fail_message => 'manifest.json file not found.');
+    validate_script_output("file $workdir/version",
+        sub { $_ !~ m/cannot/ },
+        fail_message => 'version file not found.');
+    # $_ !~ if it finds string 'cannot', test fails
+
+    ######### Run inspect locally
+
+    # skopeo inspect dir:$workdir
+    assert_script_run("skopeo inspect dir:$workdir",
+        fail_message => 'Failed to inspect local image.');
+
+    # skopeo inspect --raw dir:/tmp/test   <---- inspects raw manifest or configuration
+    assert_script_run("skopeo inspect --raw dir:$workdir",
+        fail_message => 'Failed to inspect local image.');
+
+    # skopeo inspect --raw dir:/tmp/test | jq '.config.digest'
+    # use jq to extract the value of .config.digest field from raw JSON data
+    assert_script_run("skopeo inspect --raw dir:$workdir | jq '.config.digest'",
+        fail_message => 'Failed to inspect local image.');
+
+
+
+    ######### Copy tests
+
+    # Copy from remote to dir1, to dir2;
+    # Compare dir1 and dir2, expect no changes.
+
+    # dir1 = $workdir/dir1
+    my $dir1 = "$workdir/dir1";
+    # dir2 = $workdir/dir2
+    my $dir2 = "$workdir/dir2";
+
+    # Copy from remote to dir1
+    # Combine mkdir and skopeo in a single line to account for missing directory:
+    assert_script_run("mkdir -p $dir1 && skopeo copy $remote_image dir:$dir1",
+        fail_message => 'Failed to copy remote image to directory');
+
+    # Copy from dir1 to dir2
+    assert_script_run("mkdir -p $dir2 && skopeo copy dir:$dir1 dir:$dir2",
+        fail_message => 'Failed to copy local image from one directory to another.');
+
+    # Both extracted copies must be identical.
+    # diff flags: -u is human readable format, -r compare subdirectories, N treats all files as text.
+    # assert_script_run will return 0 if there is no difference and 1 if there are differences.
+    assert_script_run("diff -urN $dir1 $dir2",
+        fail_message => 'Copied images are not identical.');
+
+}
+
```

Comment:
I would recommend adding a cleanup routine, see e.g. https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/19115/files#diff-3b34101ea0df2e75ea46097ce9ac2b57880f89129ee82472f20b3e689d6e4132R87 to remove all the temporary stuff.

Reply:
Done!